google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/ProjectPath.java

```diff
@@ -16,17 +16,35 @@
 
 package com.google.cloud.pubsublite;
 
-import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.cloud.pubsublite.internal.Preconditions.checkArgument;
 
 import com.google.auto.value.AutoValue;
+import io.grpc.StatusException;
 import java.io.Serializable;
 
 @AutoValue
 public abstract class ProjectPath implements Serializable {
-  public abstract String value();
+  public abstract ProjectIdOrNumber project();
 
-  public static ProjectPath of(String value) {
-    checkArgument(!value.isEmpty());
+  @Override
+  public String toString() {
+    return "projects/" + project();
+  }
+
+  public static ProjectPath of(ProjectIdOrNumber value) {
     return new AutoValue_ProjectPath(value);
   }
+
+  public static ProjectPath parse(String path) throws StatusException {
```

Comment:
IIUC, this is safe because project names cannot be just numbers. Is this right? I'd add a comment here.

Reply:
Project ids cannot be just numbers, correct. Added.